兼容python2 和 python3

enjoecn opened the pull request:
第一次尝试python2和python3的兼容，这个pull request，就是解决了tests下的
1. print, reduce 的 import问题
2. has_key 替换为 in

使用six来做兼容.

enjoecn committed:
1. test_*.py: has_key 修改为 in; 2. framework.py: print, reduce 的 import

liluo commented:
:cool:  非常感谢, 稍后来看~

enjoecn committed:
fix: from __future__ imports 必须位于文件最前面. 若不存在, 在python3.3.3下执行会报错SyntaxError

enjoecn commented:
@liluo 不知道是否弄对了，如果不对，告诉我，再改改。
另外，对于python3完全不兼容python2的部分，比如 input， 怎么搞？

liluo commented:
@daydayfree 帮看一下吧

enjoecn committed:
修改except后的','为as

daydayfree commented:
@Jonwei @liluo 我看修改的部分是与测试相关的，我也没有在 Python3 环境上测试，需要兼容的地方应该在 API 部分吧。
当前分支没有跟上游同步，同步下来再测试一下吧？

enjoecn commented:
@daydayfree @liluo 
1. 同步douban的分支？ 拉了douban的官方分支, 使用git diff将官方分支与本地master分支(拉得liluo的分支)对比，没有差别。当前分支是基于本地master分支创建修改的。
2. 问下，这个修改是会自己写兼容处理，还是使用six？

enjoecn committed:
1. douban_client/client.py: 显式相对import
2. tests/framework.py: 使用lib six的print_, input, reduce 兼容py2-3
3. tests/test_api_miniblog.py: 使用lib six的text_type兼容py2-3

enjoecn committed:
requirments.txt: 增加six

enjoecn commented:
1. 修改了下，采用six来做py2-3的兼容
2. 使用2to3测试了下API代码，没有提示需要修改， 所以API几乎没动.
3. 在python3.3.3环境下跑了tests/run.py.

liluo commented:
@Jonwei 
- 需要同步这个仓库的更新
- 把 `setup.py,  requirments.txt` 中 `py-oauth2>=0.0.5` 改成 `py-oauth2>=0.0.7`

enjoecn committed:
同步liluo分支, 消除冲突文件.

Merge branch 'master' into 2to3

Conflicts:
	douban_client/client.py
	tests/test_api_note.py

enjoecn committed:
requirements.txt: py-oauth2最低版本修改为0.0.7

enjoecn committed:
fix bug: DoubanApi -> DoubanAPI

enjoecn commented:
@liluo 
1.
需要同步这个仓库的更新 
把 setup.py, requirments.txt 中 py-oauth2>=0.0.5 改成 py-oauth2>=0.0.7
--- Done

2.
使用2to3测试了下API代码，没有提示需要修改， 所以API几乎没动.

3.
在python3.3.3环境下跑了tests/run.py， 无适合的code, 结果：
Ran 115 tests in 54.537s, FAILED (failures=2, errors=62)

enjoecn committed:
使用flake8检查pep8错误时, 发现以下遗漏:
 1. test_api_album.py: line42, key 应为 ret
 2. test_api_note.py : has_key 该为 in 实现

liluo commented:
@Jonwei :+1:  辛苦了, 我找台 vps 晚上跑一下.

liluo commented:
@Jonwei 

更新一下进度.

PR 已 merge,  另外 @daydayfree 也完成了一部分的工作, 目前还有 unittest 中 2 个 case 有 `need_permission`, 预计会在明天完成.
